likecoin/admin/restful.php

```diff
@@ -45,6 +45,181 @@ function likecoin_rest_refresh_publish_status( $request ) {
+	foreach ( $images as $image ) { // only works after attachment is converted to image by user.
+		// phpcs:disable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$parent = $image->parentNode;
+		if ( 'figure' === $parent->nodeName ) {
+			// phpcs:enable WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$classes = $parent->getAttribute( 'class' );
+			$parent->setAttribute( 'class', $classes . ' image' );
+		} else {
+			$figure = $dom_document->createElement( 'figure' );
+			$figure->setAttribute( 'class', 'image' );
+			$image = $parent->replaceChild( $figure, $image );
+			$figure->appendChild( $image );
+			$parent = $figure;
+		}
+		$url    = $image->getAttribute( 'src' );
+		$urls[] = $url;
+	};
+	return $urls;
+}
+/**
+ * Add likecoin arweave estimate endpoint.
+ *
+ * @param WP_REST_Request $request Full data about the request.
+ * @return WP_Error|WP_REST_Response
+ */
+function likecoin_rest_post_arweave_estimate( $request ) {
+	$post_id = $request['id'];
+	$post    = get_post( $post_id );
+	if ( ! isset( $post ) ) {
+		return new WP_Error( 'post_not_found', __( 'Post was not found', LC_PLUGIN_SLUG ), array( 'status' => 404 ) );
+	}
+	// phpcs:disable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$boundary = base64_encode( wp_generate_password( 24 ) );
+	// phpcs:enable WordPress.PHP.DiscouragedPHPFunctions.obfuscation_base64_encode
+	$body                      = likecoin_transform_content_to_arweave_accepted_form( $boundary, $post );
+	$likecoin_api_estimate_url = 'https://api.like.co/api/arweave/estimate';
+	$response                  = wp_remote_post(
+		$likecoin_api_estimate_url,
+		array(
+			'headers' => array(
+				'Content-Type' => 'multipart/form-data; boundary=' . $boundary,
+			),
+			'body'    => $body,
+		)
+	);
+	if ( is_wp_error( $response ) ) {
+		$err_message = $response->get_error_message();
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			// phpcs:disable WordPress.PHP.DevelopmentFunctions
+			error_log( $response->get_error_message() );
+			// phpcs:enable WordPress.PHP.DevelopmentFunctions
+		}
+		return array( 'error' => $response->get_error_message() );
+	}
+	$decoded_response = json_decode( $response['body'], true );
+	if ( ! $decoded_response ) {
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG === true ) {
+			// phpcs:disable WordPress.PHP.DevelopmentFunctions
+			error_log( $response['body'] );
+			// phpcs:enable WordPress.PHP.DevelopmentFunctions
+		}
+		return array( 'error' => $response['body'] );
+	}
+	$data = json_decode( $response['body'], true );
+	return new WP_REST_Response( $data, 200 );
+}
+/**
+ * Transform content to arweave-accepted body format.
+ *
+ * @param string| $boundary Random-generated separator.
+ * @param object| $post WordPress post object.
+ */
+function likecoin_transform_content_to_arweave_accepted_form( $boundary, $post ) {
+	$content        = likecoin_filter_matters_post_content( $post );
```

Comment:
Is calling `likecoin_filter_matters_post_content` desired for arweave upload?
What does it do for arweave?

Reply:
actually do not need this, will change this line

Reply:
Changed to just 
`$content        = apply_filters( 'the_content', $post->post_content );`